adaptive_memory_v4.0.py

```
@@ -0,0 +1,2119 @@
+        )
+        def check_threshold_float(cls, v, info):
+            if not (0.0 <= v <= 1.0):
+                raise ValueError(
+                    f"{info.field_name} must be between 0.0 and 1.0. Received: {v}"
+                )
+            return v
+
+        @field_validator("retry_delay")
+        def check_non_negative_float(cls, v, info):
+            if not isinstance(v, float) or v < 0.0:
+                raise ValueError(f"{info.field_name} must be a non-negative float")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)